cpp/include/cudf/detail/utilities/element_argminmax.cuh

```diff
@@ -37,7 +37,7 @@ struct element_argminmax_fn {
   bool const has_nulls;
   bool const arg_min;
 
-  __device__ inline auto operator()(size_type const& lhs_idx, size_type const& rhs_idx) const
+  __noinline__ __device__ auto operator()(size_type const& lhs_idx, size_type const& rhs_idx) const
```

Comment:
What is the difference between `__noinline__` and `__attribute__((noinline))`?

Reply:
Nothing as far as I can tell.
Also, reference: https://github.com/NVIDIA/thrust/issues/1344#issuecomment-1164676122

Reply:
I asked @jrhemstad about this once and he gave a nice example of exactly how equivalent they are once nvcc is done preprocessing the files:
```
(rapids) rapids@compose:~/cudf/tmp$ echo "__global__ void kernel(){} int main(){}" > test.cu && nvcc ./test.cu --keep && tail -3 test.cpp1.ii
# 1 "<command-line>" 2
# 1 "./test.cu"
__attribute__((global)) void kernel(){} int main(){}
```

Reply:
So `__inline__` will be converted into `__attribute__((inline))` right? I guess so, as `__something__` seems a CUDA specific keyword and will be converted into some C++ standard equivalent if possible.

Reply:
I was wrong:
```
╰─ echo "__inline__ __device__ void f(){} int main(){}" > test.cu && nvcc ./test.cu --keep && tail -3 test.cpp1.ii
# 0 "<command-line>" 2
# 1 "./test.cu"
__inline__ __attribute__((device)) void f(){} int main(){}
```

Reply:
When we use `__noinline__` for a specific reason like this that is affected by external code (CUB), we should add a comment to indicate why.
```suggestion
  // Must be __noinline__ for Thrust/CUB, to prevent a long compile time
  __noinline__ __device__ auto operator()(size_type const& lhs_idx, size_type const& rhs_idx) const
```

Reply:
Already had it queued up. :+1: